packages/jest-runner/src/testWorker.ts

```diff
@@ -74,7 +74,10 @@ export function setup(setupData: {
     config,
     serializableModuleMap,
   } of setupData.serializableResolvers) {
-    const moduleMap = ModuleMap.fromJSON(serializableModuleMap);
+    const HasteMapClass = config.haste.hasteMapModulePath
```

Comment:
code in this general area is what's causing https://github.com/facebook/jest/issues/9021#issuecomment-825410785 - you wouldn't by any chance be able to take a look at that while you're touching this part of the code? 😀